File: read_glb.py
import json, struct, os

path = r'e:\Phase 1_ Quantum Foundation Project Setup Instructions\frontend\public\assets\office.glb'
size = os.path.getsize(path)
print(f'File size: {size/1024/1024:.1f} MB')

with open(path, 'rb') as f:
    magic, ver, length = struct.unpack('<III', f.read(12))
    print(f'Magic: {magic:#010x}, Version: {ver}')
    chunk_len, chunk_type = struct.unpack('<II', f.read(8))
    json_bytes = f.read(chunk_len)
    gltf = json.loads(json_bytes)

meshes = gltf.get('meshes', [])
nodes  = gltf.get('nodes', [])
mats   = gltf.get('materials', [])

print(f'\nMeshes ({len(meshes)}):')
for m in meshes:
    print(f'  {m.get("name","unnamed")}')

print(f'\nMaterials ({len(mats)}):')
for mat in mats:
    pbr = mat.get('pbrMetallicRoughness', {})
    base = pbr.get('baseColorFactor', None)
    print(f'  {mat.get("name","unnamed")}  base={base}')

print(f'\nNodes with mesh ({len([n for n in nodes if "mesh" in n])}):')
for n in nodes:
    if 'mesh' in n:
        mesh_name = meshes[n['mesh']].get('name','unnamed') if n['mesh'] < len(meshes) else '?'
        print(f'  node: {n.get("name","unnamed")} -> mesh: {mesh_name}')

File: read_glb.mjs
import { readFileSync } from 'fs';

const path = 'frontend/public/assets/office.glb';
const buf = readFileSync(path);

const size = buf.length;
console.log(`File size: ${(size/1024/1024).toFixed(1)} MB`);

// GLB header: magic(4) version(4) length(4)
const magic = buf.readUInt32LE(0);
const ver   = buf.readUInt32LE(4);
console.log(`Magic: 0x${magic.toString(16)}, Version: ${ver}`);

// Chunk 0 header: chunkLength(4) chunkType(4)
const chunkLen  = buf.readUInt32LE(12);
const chunkType = buf.readUInt32LE(16);
const jsonStr   = buf.subarray(20, 20 + chunkLen).toString('utf8');
const gltf      = JSON.parse(jsonStr);

const meshes = gltf.meshes  || [];
const nodes  = gltf.nodes   || [];
const mats   = gltf.materials || [];

console.log(`\nMeshes (${meshes.length}):`);
meshes.forEach(m => console.log(`  "${m.name || 'unnamed'}"`));

console.log(`\nMaterials (${mats.length}):`);
mats.forEach(mat => {
  const pbr  = mat.pbrMetallicRoughness || {};
  const base = pbr.baseColorFactor;
  console.log(`  "${mat.name || 'unnamed'}"  base=${JSON.stringify(base)}`);
});

const meshNodes = nodes.filter(n => n.mesh !== undefined);
console.log(`\nNodes with mesh (${meshNodes.length}):`);
meshNodes.forEach(n => {
  const meshName = n.mesh < meshes.length ? meshes[n.mesh].name : '?';
  console.log(`  node: "${n.name || 'unnamed'}" -> mesh: "${meshName}"`);
});
